BESTPRACTICES.md

```diff
@@ -0,0 +1,18 @@
+# Best Practices
+---
+## Weave 
+> Is weave.Metadata metadata always required?
+
+This allows you to upgrade the code on chain without downtime
```

Comment:
Add link to `weave/migrations`

Reply:
The docs.go file in there is actually quite insightful, probably best to link directly to it